sdk/tests/core/test_az_json.c

```diff
@@ -3068,16 +3102,24 @@ static void test_az_json_string_unescape(void** state)
     destination_span._internal.size = final_size;
```

Comment:
Nit: Rather than manually modifying the internals of the az_span, which shouldn't be used outside the type, use the public slice API (here and elsewhere).

```suggestion
    destination_span = az_span_slice(destination_span, 0, final_size):
```

Just a nit, since it's test code.